### DISCUSSION TITLE
Nextjs V9 Dynamic routes 404 when using `now dev` but not `next`

### DISCUSSION THREAD

**Discussion by pkearney13** [vercel/next.js#12552] [Help] [answered]
Hi there,

I am looking for some help with an issue I am facing. I am building a small application that uses Nextjs V9 and I am using `now` CLI for deploying and for local development.

I am trying to create a simple profile page that has the route `http://localhost:3000/profile/id`

I have followed so many guides and all of them suggest to use the following different page setups `pages/profile/[id]/index.js`
`pages/profile/[...slug.js]`
`pages/profile/[idj.s]`

I have used all of the above in trying to create a route where I can access the route params. I can use query params but I really want to try and have it like 

`http://localhost:3000/profile/123`

Rather than 

`http://localhost:3000/profile?id=123`

It seems to work fine when I use `next` CLI but not `now dev` CLI I just get a 404. 

Am I missing something obvious with how `next` and `now dev` differ? 

If I have forgotten any information that someone might need to help please ask and I will provide. 

Thanks in advance.

**Comment by paulogdm** [comment]
Can you provide us with a minimal reproducible example?

`now dev` should have the same behavior of `next dev` as of today.

**Comment by pkearney13** [comment, marked as answer]
Closing this as it turns out I needed to configure routes inside a now.json file.
